tests.py

```diff
@@ -81,5 +104,33 @@ def test_subclauses_arithmetic_booleans(self):
         self.run_vsm('subclauses_arithmetic_booleans', input_data,
                 expected_output, False)
 
+
+class TestIPC(TestVSM):
+
+    def setUp(self):
+        load_plugin('ipc.{}'.format(IPC_MODULE))
+
+    def test_simple0(self):
+        input_data = [('transmission_gear', '"reverse"')]
+        expected_output = ('car.backup', 'True')
+        self.run_vsm('simple0', input_data, expected_output)
+
+    def test_simple2_initial(self):
+        input_data = [('damage', 'true')]
+        expected_output = ('car.stop', 'True')
+        self.run_vsm('simple2', input_data, expected_output)
+
+    def test_simple2_multiple_signals(self):
+        input_data = [('moving', 'false'), ('damage', 'true')]
+        expected_output = ('car.stop', 'True')
+        self.run_vsm('simple2', input_data, expected_output, False)
+
+
 if __name__ == '__main__':
-    unittest.main()
+    IPC_MODULE=None
```

Comment:
So what I was proposing was setting the flag as a variable on the class. So you'll do something like this:

```
TestVSM.use_ipc_module = None
unittest.main()
TestVSM.use_ipc_module = 'zeromq'
unittest.main()
```
The first call will run the tests with the default input/output, the second will use zeromq. Then in run_vsm you'll have:

```
if self.use_ipc_module:
   return self.run_with_ipc(…)
```

And in that function you can deal with all of the special cases such as breaking up the input string, putting the tuples back together, so that they can be properly compared.

For instance, this will take the input_data string the existing tests have and turn them into your list of tuples:

```
if self.use_ipc_module = 'zeromq':
    input_str = input_data
    input_data = []
    for line in input_str.strip().split('\n'):
        operands = map(str.strip, line.split('='))
        input_data.append(tuple(operands))
```

Let me know if you need help with that?